pkg/diagnose/firewall.go

```diff
@@ -295,3 +311,29 @@ func getTargetPort(submariner *v1alpha1.Submariner, endpoint *subv1.Endpoint, tg
 		return 0, fmt.Errorf("could not determine the target port for cable driver %q", endpoint.Spec.Backend)
 	}
 }
+
+func getLbNodePort(clusterInfo *cluster.Info, endpoint *subv1.Endpoint, tgtport TargetPort) (int32, error) {
+	usingLoadBalancer, _ := endpoint.Spec.GetBackendBool(subv1.UsingLoadBalancer, nil)
+	if usingLoadBalancer == nil || !*usingLoadBalancer {
+		return 0, nil
+	}
+
+	portName := encapsPortName
+	if tgtport == NatDiscoveryPort {
+		portName = nattPortName
+	}
+
+	svc, err := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Services(endpoint.GetNamespace()).Get(
+		context.TODO(), loadBalancerName, metav1.GetOptions{})
+	if err == nil {
+		for _, port := range svc.Spec.Ports {
+			if port.Name == portName {
+				return port.NodePort, nil
+			}
+		}
+	} else {
+		return 0, fmt.Errorf("error reading the details of LB service %s: %w", loadBalancerName, err)
+	}
+
+	return 0, fmt.Errorf("could not determine nodePort for port name %q of LB service %s", portName, loadBalancerName)
```

Comment:
Is it an error if it's not found?

Reply:
For the platforms I tested (aws and gcp) nodePort was created for each LB port, I'm pretty sure Nodeport should always be there, so I guess we should report an error if we can not find nodePort.